contracts/eth/nearbridge/contracts/NearBridge.sol

```diff
@@ -21,33 +21,35 @@ contract NearBridge is INearBridge, AdminControlled {
         uint256 stakeThreshold;
     }
 
-    // Whether the contract was initialized.
-    bool public initialized;
     uint256 public lockEthAmount;
     // lockDuration and replaceDuration shouldn't be extremely big, so adding them to an uint64 timestamp should not overflow uint256.
     uint256 public lockDuration;
     // replaceDuration is in nanoseconds, because it is a difference between NEAR timestamps.
     uint256 public replaceDuration;
     Ed25519 immutable edwards;
 
-    Epoch[3] epochs;
-    uint curEpoch;
-    uint64 curHeight;
+    // End of challenge period. If zero, untrusted* fields and lastSubmitter are not meaningful.
+    uint256 public lastValidAt;
 
+    uint64 curHeight;
     // The most recently added block. May still be in its challenge period, so should not be trusted.
     uint64 untrustedHeight;
-    uint256 untrustedTimestamp;
+
+    // Address of the account which submitted the last block.
+    address lastSubmitter;
+
+    // Whether the contract was initialized.
+    bool public initialized;
     bool untrustedNextEpoch;
     bytes32 untrustedHash;
     bytes32 untrustedMerkleRoot;
     bytes32 untrustedNextHash;
+    uint256 untrustedTimestamp;
```

Comment:
It is better to group all the variables with the same slot size (256) together. I know it is also confusing the state variables readability ... but in general it is ok

Reply:
yeah, I agree on that in general. In this specific case, as the contract is quite complex I strive more for having readability too and at least somehow group public/private fields. I know that this `initialized` is out of the rule here too, as an exception. As grouping uint256 makes no sense for the performance optimization, I'd leave it as-is for readability purposes.